src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs

```diff
@@ -56,37 +56,71 @@ public abstract class NoxDatabaseConfigurator : INoxDatabaseConfigurator
             Entity entity,
             IReadOnlyList<EntityRelationshipWithType> relationshipsToCreate)
         {
+            
             foreach (var relationshipToCreate in relationshipsToCreate)
             {
-                // ManyToMany does not need to be handled
-                // Handle ZeroOrOne or ExactlyOne scenario with foreign key.
-                if (relationshipToCreate.Relationship.HasRelationshipWithSingularEntity() &&
-                    relationshipToCreate.Relationship.ShouldGenerateForeignOnThisSide() &&
-                    !relationshipToCreate.Relationship.IsManyRelationshipOnOtherSide())
+                // One to ?? (// Many to Many are setup by EF)
+                if (relationshipToCreate.Relationship.ShouldGenerateForeignOnThisSide() && relationshipToCreate.Relationship.WithSingleEntity())
                 {
-                    builder
-                        .HasOne(relationshipToCreate.Relationship.Entity)
-                        .WithOne(entity.Name)
-                        .HasForeignKey(entity.Name, $"{relationshipToCreate.Relationship.Entity}Id");
-
-                    // Setup one to one foreign key
+                    //One to One
+                    if (!relationshipToCreate.Relationship.IsManyRelationshipOnOtherSide())
```

Comment:
can be reversed to not have negative branch here

Reply:
done